subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -3055,15 +3035,15 @@ int lwm2m_engine_send(struct lwm2m_ctx *ctx, char const *path_list[], uint8_t pa
 
 msg_init:
 
-	if (confirmation_request) {
-		msg->type = COAP_TYPE_CON;
-		msg->reply_cb = do_send_reply_cb;
-		msg->message_timeout_cb = do_send_timeout_cb;
-	} else {
-		msg->type = COAP_TYPE_NON_CON;
-		msg->reply_cb = NULL;
-		msg->message_timeout_cb = NULL;
+	if (!reply_cb) {
+		LOG_ERR("Undefined reply callback");
+		return -EFAULT;
 	}
```

Comment:
This should not be an error. Instead if no callback is provided, use the default one.

Reply:
Done